system/jlib/jstats.h

```diff
@@ -1044,4 +1046,94 @@ inline double calcCost(double ratePerHour, unsigned __int64 ms) { return ratePer
 inline double calcCostNs(double ratePerHour, unsigned __int64 ns) { return ratePerHour * ns / 1000000000 / 3600; }
 extern jlib_decl StringBuffer & formatMoney(StringBuffer &out, unsigned __int64 value);
 
+/*
+ * ThresholdMetric
+ *
+ * Standalone metric type designed for tracking numeric values such as transaction latency,
+ * response times, or other measurements where statistical analysis is needed.
+ *
+ * This metric uses Welford’s online algorithm for running mean and variance, providing
+ * numerically stable and efficient calculation of average and standard deviation without
+ * storing historical data. It supports outlier detection via deviation calculations and
+ * can be cleared to reset all statistics.
+ */
+
+/*
+ * Performance Design Notes:
+ * - Uses RelaxedAtomic<__uint64> for count to enable lock-free queryCount() operations
+ * - recordValue(double) accepts native floating-point values without conversion overhead
+ * - Critical sections protect both count increment and mean/M2 updates to ensure Welford's
+ *   algorithm correctness - count and statistical updates must be applied in sequential order
+ * - Supports implicit conversion from integer types for caller convenience
+ * - Memory layout groups related fields together without forced alignment since mean/M2
+ *   are always accessed together under the same lock
+ *
+ * Thread Safety:
+ * - queryCount() is lock-free and safe for concurrent access
+ * - recordValue() uses critical section to ensure sequential application of Welford updates
+ * - Other query operations use minimal critical sections for mathematical correctness
+ * - Multiple threads can safely record values and query statistics concurrently
+ */
+class ThresholdStat
+{
+public:
+    ThresholdStat() {}
+
+    __uint64 queryCount() const  // Lock-free operation using RelaxedAtomic
+    {
+        return count.load();
+    }
+
+    void recordValue(double value)  // Accepts double directly, supports implicit conversion from integers
+    {
+        CriticalBlock block(cs);  // Ensure atomic count increment with mean/M2 updates
+        __uint64 newCount = count.add_fetch(1);  // Atomic increment inside critical section
```

Comment:
Minor: Since this is only updated inside a critical section you can use count.fastAdd(1) - which is more efficient.